### ISSUE TITLE
Use %pythonabc without -py3

### ISSUE THREAD

**Issue by msteinbeck** [swig/swig#482]
Lets say I have the class `MyClass` and I want it to extend `collections.MutableSequence`. According to the [documentation](http://www.swig.org/Doc1.3/Python.html#Python_nn76) I have to do the following:

```
%include <pyabc.i>
%pythonabc(MyClass, collections.MutableSequence);
...
swig -python -py3 ...
```

If I use -py3 everything works as expected. If I do not use -py3 `MyClass` extends `object` instead of `collections.MutableSequence`. Actually subclassing `collections.MutableSequence` is feasible with  python 2. So why do I have to use -py3 if I want to extend a class with %pythonabc?

**Comment by wsfulton** [comment]
I suppose Python Abstract Base Classes, as documented here, https://www.python.org/dev/peps/pep-3119/ were designed for Python 3. Were they backported to 2? If so, then we can remove the -py3 restriction. 

The patch to do this is simply (untested): 

```patch
diff --git a/Source/Modules/python.cxx b/Source/Modules/python.cxx
index 5670d95..36d759e 100644
--- a/Source/Modules/python.cxx
+++ b/Source/Modules/python.cxx
@@ -4293,7 +4293,7 @@ public:
 
       /* dealing with abstract base class */
       String *abcs = Getattr(n, "feature:python:abc");
-      if (py3 && abcs) {
+      if (abcs) {
        if (Len(base_class)) {
          Printv(base_class, ", ", NIL);
        }
```

Rename Examples/test-suite/python/python_abstractbase_runme3.py  to Examples/test-suite/python/python_abstractbase_runme.py and see if the test-suite passes for all the combinations of Python 2.

**Comment by ojwb** [comment]
>Were they backported to 2?

Yes, to Python 2.6 according to:

https://python.readthedocs.io/en/v2.7.2/library/abc.html

I'll try a PR based on the patch above.

**Comment by ojwb** [comment]
Oh, except from `CHANGES`:

```
2018-06-12: wsfulton
            [Python] Python 3.7 support: The %pythonabc feature in pyabc.i now uses base classes
              collections.abc.MutableSequence
              collections.abc.MutableMapping
              collections.abc.MutableSet
            instead of
              collections.MutableSequence
              collections.MutableMapping
              collections.MutableSet
            as the latter are deprecated in Python 3.7 and are due to be removed in Python 3.8.
            The classes in collections.abc.* are available from Python 3.3 onwards. If you
            require support for Python 3.2, then copy the pyabc.i file and modify by removing
            the few instances of the .abc sub-module.
```

So since 4.0.0 we only actually support this feature for Python 3.3 and later.

Back in 2015 supporting this for Python 2 would have been worthwhile, but Python 2 reached end of life upstream over two years ago (2020-01-01) so it doesn't make sense to put non-trivial effort into supporting new features for it at this point in time and I'm going to just close this as "wontfix".
